cluster/sync.sh

```diff
@@ -16,4 +16,9 @@ rm -rf $destination
 mkdir -p $destination
 DESTINATION=$destination IMAGE_REGISTRY=registry:5000 make manifests
 
-# TODO: Remove old components and apply new ones
+./cluster/kubectl.sh delete --ignore-not-found configmap macvtap-deviceplugin-config
+./cluster/kubectl.sh delete --ignore-not-found ds macvtap-cni
+
+./cluster/kubectl.sh create -f examples/macvtap-deviceplugin-config.yaml
```

Comment:
We're not templating the DP at all - nor have it available in _out.

While I agree it's odd to deploy the cluster using the 'examples' folder, I don't have a better alternative. It feels a waste of time to copy the example to the _out or even template it.

I think the best choice is to get rid of the config map altogether & simply rely on the NAD for this.

Reply:
It is configuration and will definitely not apply to all deployments (i.e. not all deployments will have an eth0 interface). That's is why it is provided as an example. It does apply to the kubevirt CI so I don't see a real problem to be used it in here. I do see a case to apply it on the test itself rather than here, just because someone might want to run the environment and configure macvtap a different way and this specific configuration is just needed to run the test.

> I think the best choice is to get rid of the config map altogether & simply rely on the NAD for this.

What do you mean? I don't see how this could be done.

Reply:
Have the device plugin listen to NAD provisioning & re-acting accordingly.

Anytime it seems a NAD w/ for a macvtap resource, it would update its internal configuration.

I think this is feasible. Do you see any reason why it won't be ?

Reply:
It just doesn't make any sense providing and consuming a resource from the same place. We need the separation of concern, admins provide host resources, users consume them.

Reply:
NetworkAttachmentDefinition is a resource that is created only by admins. By creating a NAD you can get access to any network in the cluster. Nobody but admin should be able to do it.

So the proposed mechanism is to let admin create NetworkAttachmentDefinition to expose "an option to attach to interface X" to users.

Reply:
Still does not make sense to me providing and consuming from the same place, I see how it is technically feasible and maybe appealing for the config update use case, but it opens more questions that it answers for me. Let's have an issue to discuss this, I don't think it should be relevant to this PR.

Reply:
You are right, it is not related to this PR and is worth a separate discussion.

Reply:
> Still does not make sense to me providing and consuming from the same place, I see how it is technically feasible and maybe appealing for the config update use case, but it opens more questions that it answers for me. Let's have an issue to discuss this, I don't think it should be relevant to this PR.

I agree that this is unrelated to this PR, and that we should discuss it elsewhere.

Please open an issue w/ your questions, because up to now, I don't understand your point of view - I think Petr answered the only concern you've mentioned.

Reply:
@jcaamano thanks for opening https://github.com/kubevirt/macvtap-cni/issues/8 .